ADRs/ADR_AI_REFINE_PROMPT.md

```diff
@@ -0,0 +1,24 @@
+Consider the following documents and conversations:
```

Comment:
I just realized that we can build a service that helps designers / architects to work on a design doc.

```
User --> Service --> OpenAI
                loop
			Service --> OpenAI
			Service <-- OpenAI
		end
User <-- Service <-- OpenAI
```

The Service would have a collection of questions that we pre-pare, for example:
- Check if the user has defined the problem clearly
- Check if the user has provided sufficient quantifiable evidence
- Check if the writing is clear
- Check if there is an easy-to-read introduction and a concise summary

So if the user forgets or is inexperienced enough to ask for the right feedback, we can do it for them.

Wdyt!? I think we can even do (hopefully together) as an iteration on this. We can just build the service do it. We can get GPT to do it. I say we should just fucking do it. 🤝 ?

Reply:
https://til.simonwillison.net/llms/python-react-pattern

I learnt that this is called the ReAct pattern (Reason + Act) in the context of LLMs: https://til.simonwillison.net/llms/python-react-pattern

I think we can really do some meaningful that helps us with design doc design - this can be far more reaching than just pocket.

Will keep ideating on the idea from last week.